### ISSUE TITLE
Support scaling the UI based on terminal size

### ISSUE THREAD

**Issue by ristomatti** [dhth/hours#10]
**Describe the solution you'd like**

The position and width of the columns to scale based on terminal width.

**Is your feature request related to a problem? Please describe.**

I use a tiling window manager on an ultrawide monitor. I usually keep a column of relatively narrow (~80 character) terminals on one side of the monitor for secondary information. While `hours` is perfectly usable with 80 char width, it bugs me that the "worked on for" column isn't visible, even though there'd be plenty of room (with short task titles at least).

Currently the Task List View requires around 90 character wide window to fit. The Task Log Entry view requires 115 character wide terminal to fit in all columns. At this width, it unnecessarily cuts the description shorter than the available space. The third column on the other hand is unnecessarily far apart from the second one.

**Describe alternatives you've considered**

N/A

**Additional context**

In case relevant: I'm using Ubuntu with i3 window manager and kitty as the terminal.

Task List View on a 80 character wide terminal:

![image](https://github.com/user-attachments/assets/914b3ad4-8bf6-4b56-8a21-393859461112)

Task Log Entry View on a 115 character wide terminal:

![image](https://github.com/user-attachments/assets/59d95f7c-c880-4a75-bc3d-59c3dc34d1c7)

**Comment by ristomatti** [comment]
Aside: I'm currently testing [omm](https://github.com/dhth/omm/) also. I noticed it also seems to have the same issue. I hope it's enought to report this here only as both make use  UI component libraries and look very much alike. :slightly_smiling_face:

**Comment by dhth** [comment]
I've been meaning to do that. Since you already tried out `omm`, could I ask you to take a quick look at https://github.com/dhth/prs, and see if its responsive layout is in line with what you described above?

**Comment by ristomatti** [comment]
Yup that would fill the need for `hours` or `omm` as they only have few columns.

I have a few suggestions on how to further improve the UX in case you happen to be interested in experimenting. I'll use `prs` as an example since it has more information to display. Also, when I first run it, I realized I seem to have some pretty ancient open PR's that have both long titles and target repository names. 🙂

I'll let my MS Paint style annotations speak for themselves.... 

![image](https://github.com/user-attachments/assets/b08760d6-6db2-47d6-999b-28cdc48f1407)

**Comment by ristomatti** [comment]
Other suggestions that might be more work to implement:
4. Wrap long PR tittles instead of truncating
5. On small windows(*), display the fields as a list instead of columns

**Comment by ristomatti** [comment]
One more suggestion. It might be preferable to to just refuse to render at some point instead of trying to fit everything when the terminal size shrinks past a specified limit:

![image](https://github.com/user-attachments/assets/5b962b14-0006-4748-875f-80d060215ad2)

Here's a screenshot of [btop](https://github.com/aristocratos/btop) doing just that:

![image](https://github.com/user-attachments/assets/8bfd0cbe-070d-4077-86dd-d35b63d30d68)

**Comment by dhth** [comment]
Thanks for the great suggestions 🙂
I'll definitely incorporate the following:

- refuse to render when width/height shrinks below a threshold
- better use of spacing when the window size is large enough

I'll experiment a bit with the other ones.
I've also opened an issue for this for `prs` [here](https://github.com/dhth/prs/issues/13).
